### PR TITLE
discord: fix updater script

### PR DESCRIPTION
**Pull request by mohe2015**
###### Motivation for this change
The update script produced the wrong commit message. This should be fixed now.
Closes https://github.com/NixOS/nixpkgs/pull/128594

###### Things done
The download url is minimally modified but I tested it and it still works like before.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mohe2015** [comment]
Result of `nixpkgs-review pr 128633` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discord-canary</li>
  </ul>
</details>

**Comment by mohe2015** [comment]
`NIXPKGS_ALLOW_UNFREE=1 nix shell --impure github:mohe2015/nixpkgs/fix-discord-autoupdate#discord-canary` seems to run. (Just logged in and read a chat)

File: pkgs/applications/networking/instant-messengers/discord/base.nix [hunk 1]
```diff
@@ -71,7 +72,14 @@ in stdenv.mkDerivation rec {
     mimeType = "x-scheme-handler/discord";
   };
 
-  passthru.updateScript = ./update-discord.sh;
+  passthru.updateScript = writeScript "discord-update-script" ''
+    #!${stdenv.shell}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/discord/base.nix:76]:
```suggestion
    #!/usr/bin/env nix-shell
    #!nix-shell -i bash -p curl gnugrep common-updater-scripts
```

File: pkgs/applications/networking/instant-messengers/discord/base.nix [hunk 2]
```diff
@@ -71,7 +72,14 @@ in stdenv.mkDerivation rec {
...
+    set -o errexit;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/discord/base.nix:77]:
```suggestion
    set -eou pipefail;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128633` at 0e4318f2 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>discord-canary</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/instant-messengers/discord/base.nix:44:3:
```
   |
44 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Review by Artturin — APPROVED** [review]

**Comment by mohe2015** [comment]
Thanks @Artturin for your efforts.
As discord canary doesn't even seems to start any more (https://github.com/NixOS/nixpkgs/pull/130243) maybe merge soon?

File: pkgs/applications/networking/instant-messengers/discord/base.nix [hunk 3]
```diff
@@ -71,7 +72,15 @@ in stdenv.mkDerivation rec {
...
 
-  passthru.updateScript = ./update-discord.sh;
+  passthru.updateScript = writeScript "discord-update-script" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p curl gnugrep common-updater-scripts
+    set -eou pipefail;
+    url=$(curl -sI "https://discordapp.com/api/download/${builtins.replaceStrings ["discord-" "discord"] ["" "stable"] pname}?platform=linux&format=tar.gz" | grep -oP 'location: \K\S+')
+    version=''${url##https://dl*.discordapp.net/apps/linux/}
+    version=''${version%%/*.tar.gz}
+    ${common-updater-scripts}/bin/update-source-version ${pname} "$version" --file=./pkgs/applications/networking/instant-messengers/discord/default.nix
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/discord/base.nix:82]:
```suggestion
    update-source-version ${pname} "$version" --file=./pkgs/applications/networking/instant-messengers/discord/default.nix
```

**Comment by andrewrk** [comment]
So um how do we run the script now that it is deleted?

**Comment by Artturin** [comment]
> So um how do we run the script now that it is deleted?

`nix-shell maintainers/scripts/update.nix --argstr package discord`

**Comment by andrewrk** [comment]
Thanks! Worked perfectly :+1: 

I learned about how to use this general purpose feature of NixOS in the process.

**Pull request merged**